zboss/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+include(${NRFXLIB_DIR}/common.cmake)
+
+# Common includes, not related to ZBOSS source code.
+zephyr_include_directories(src/zb_error)
+zephyr_include_directories(include/addons)
+
+# Source files
+zephyr_sources(src/zb_error/zb_error_to_string.c)
+
+# Suppress bitfield compatibility warning
+zephyr_compile_options(-Wno-packed-bitfield-compat)
+
+# Add compile-time definition, indicating which ZBOSS library will be linked
+if (CONFIG_ZIGBEE_ROLE_END_DEVICE)
+  zephyr_compile_definitions(ZB_ED_ROLE)
+endif()
+
+
+if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
+    (CONFIG_ZBOSS_LIBRARY_PRECOMPILED))
+
+  zephyr_include_directories(include)
+  zephyr_include_directories(include/zcl)
+  zephyr_include_directories(include/ha)
+  zephyr_include_directories(include/se)
+  zephyr_include_directories(include/osif)
+
+  nrfxlib_calculate_lib_path(lib_path)
+  set(ZBOSS_LIB_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${lib_path})
+  if(NOT EXISTS ${ZBOSS_LIB_PATH})
+    message(WARNING "This combination of SoC and floating point ABI is not supported by the ZBOSS lib."
+                    "(${ZBOSS_LIB_PATH} doesn't exist.)")
+  endif()
+
+  if (CONFIG_ZIGBEE_ROLE_COORDINATOR OR CONFIG_ZIGBEE_ROLE_ROUTER)
+    zephyr_link_libraries(${ZBOSS_LIB_PATH}/libzboss.a)
+    toolchain_cc_imacros(${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config.h)
```

Comment:
Why do we want this as a imacro getting included everywhere ?
Should only be included by those that needs it ?
As this is config settings, would it not be possible to set a conf file in the zboss headers, similar to how mbedTLs does things.

The imacro will cause this to be used for all C-files, which should really be for really global things like the autoconf.h

Reply:
Well - it replicates exactly what is happening when you have source code with the corresponding Kconfig file, which brings us to a "unified" approach for all build types.

In precompiled builds, those defines are needed by:
 - Zigbe subsys
 - Zigbee applications (those with CONFIG_ZIGBEE == y)
 - Zigbee API test cases (those with CONFIG_ZIGBEE == y)

Additionally, this header must not be included if the ZBOSS source code is available.

If you look at those cases, you should see that the "imacro" approach is the simplest and shortest one you can imagine.

If we need to fix this, than yes, this should be pulled in via the default `#include <zboss_api.h>` directive. For example:
```
// file: zb_vendor.h
#include "libzboss_config_generic.h"
```
and - pulled in in a way, that it gets removed in source code builds:
```
// file: libzboss_config_generic.h.template
#cmakedefine LIBZBOSS_CONFIG_FILE
#ifdef LIBZBOSS_CONFIG_FILE
#include "${LIBZBOSS_CONFIG_FILE}"
#endif
```

What do you think about this approach?

Reply:
> If you look at those cases, you should see that the "imacro" approach is the simplest and shortest one you can imagine.

The shortest way is not always the safest ;)
This works when using `gcc`, but there are other toolchains out there which doesn't support imacros.
As well as the previous comment, (and strongest reason), there is no reason to expand this header when compiling Zephyr kernel. 

> What do you think about this approach?

Good direction, but I think can be reduced and still achieve same goal.
Placed some suggestions:
https://github.com/maciekfabia/nrfxlib/pull/2#pullrequestreview-412390777